src/differentials/notimplemented.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    NotImplemented
+
+This differential indicates that the derivative is not implemented.
+"""
+struct NotImplemented{M,S,I} <: AbstractDifferential
+    mod::M
+    source::S
+    info::I
+end
+
+NotImplemented() = NotImplemented(nothing, nothing, nothing)
+
+extern(x::NotImplemented) = _error(x)
+
+Base.iterate(x::NotImplemented) = _error(x)
+Base.iterate(x::NotImplemented, ::Any) = _error(x)
+
+Base.:+(x::NotImplemented, ::Any) = x
+Base.:+(::Any, x::NotImplemented) = x
+Base.:+(x::NotImplemented, ::NotImplemented) = x
+Base.:*(x::NotImplemented, ::Any) = x
+Base.:*(::Any, x::NotImplemented) = x
+Base.:*(x::NotImplemented, ::NotImplemented) = x
+
+# Linear operators
+Base.adjoint(x::NotImplemented) = x
+Base.transpose(x::NotImplemented) = x
```

Comment:
Probably don't need these two since `conj` falls back to the identical `AbstractDifferential` definition.